src/bokeh/models/widgets/tables.py

```diff
@@ -858,6 +858,50 @@ def __init__(self, *args, **kwargs) -> None:
     The height of each row in pixels.
     """)
 
+    @staticmethod
+    def from_data(data, columns=None, formatters={}, **kwargs) -> DataTable:
+        """ Create a simple table from a pandas dataframe, dictionary or ColumnDataSource.
+
+        Args:
+            data (DataFrame or dict or ColumnDataSource) :
+                The data to create the table from. If the data is a dataframe
+                or dictionary, a ColumnDataSource will be created from it.
+
+            columns (list, optional) :
+                A list of column names to use from the input data.
+                If None, use all columns. (default: None)
+
+            formatters (dict, optional) :
+                A mapping of column names and corresponding Formatters to
+                apply to each column. (default: None)
+
+        Keyword arguments:
+            Any additional keyword arguments will be passed to DataTable
+
+        Returns:
+            DataTable
+
+        Raises:
+            ValueError
+                If the data is not a pandas Dataframe, dictionary or ColumnDataSource
+
+        """
+        import pandas as pd
+
+        if isinstance(data, (pd.DataFrame, dict)):
+            source = ColumnDataSource(data)
```

Comment:
When `ColumnDataSource`' constructor changes allowing more types of input data, then we will have to find an change each case like this one. What I would do, is to check for `ColumnDataSource` first and take it as-is (well see below), and then let `ColumnDataSource()` to figure out the rest. If you want, you can catch the `ValueError` and re-throw it with a customized error message, like "Expected a ColumnDataSource or something convertible to a data source like a dict or a DataFrame".

Reply:
I agree that the suggested approach abot is more future-proofed.